arcee/api.py

```diff
@@ -23,66 +23,105 @@ def upload_corpus_folder(corpus: str, s3_folder_url: str) -> Dict[str, str]:
-
-    Returns:
-        Dict[str, str]: The response from the make_request call.
-    """
-    if len(qa_pairs) > 2000:
-        raise Exception("You can only upload 2000 QA pairs at a time")
-
-    qa_list = []
-    for qa in qa_pairs:
-        if prompt_column not in qa.keys() or completion_column not in qa.keys():
-            raise Exception("Each QA pair must have a 'question' and an 'answer' key")
-
-        qa_list.append({"question": qa[prompt_column], "answer": qa[completion_column]})
-
-    data = {"qa_set_name": qa_set, "qa_pairs": qa_list}
-    return make_request("post", Route.alignment + "/qaUpload", data)
+import os
+import csv
+import json
+import arcee
+from tqdm import tqdm
+from requests.exceptions import ConnectionError
+from datasets import load_dataset
+from typing import List, Dict
 
 def chunk_list(lst, chunk_size):
     """Yield successive n-sized chunks from lst."""
     for i in range(0, len(lst), chunk_size):
         yield lst[i:i + chunk_size]
 
-def upload_instructions_from_csv(qa_set: str, csv_path: str, prompt_column: str = "prompt", completion_column: str = "completion", batch_size: int = 200) -> None:
-    """
-    Upload QA pairs from a CSV file to a specific QA set.
-
-    Args:
-        qa_set (str): The name of the QA set to upload to.
-        csv_path (str): The path to the CSV file containing QA pairs.
-
-    Returns:
-        Dict[str, str]: The response from the make_request call.
-    """
-    qa_pairs = []
-
-    print(f"Reading QA pairs from {csv_path}...")
-    with open(csv_path, 'r', encoding='utf-8') as csvfile:
-        reader = csv.DictReader(csvfile)
-        for row in reader:
-            if prompt_column not in row.keys() or completion_column not in row.keys():
-                #raise Exception(f"Each row must have a '{question_column}' and an '{answer_column}' key. You can override the column names using the question_column and answer_column arguments.")
-                raise Exception(f"Each row must have a '{prompt_column}' and an '{completion_column}' key. You can override the column names using the prompt_column and completion_column arguments.")
-            qa_pairs.append({
-                f"{prompt_column}": row[prompt_column],
-                f"{completion_column}": row[completion_column]
-            })
-
-    print(f"Total QA pairs read: {len(qa_pairs)}")
-    # Split the QA pairs into chunks and upload each chunk separately
-    for i, chunk in enumerate(chunk_list(qa_pairs, batch_size)):
-        print(f"Uploading chunk {i + 1} of {len(qa_pairs) // batch_size + 1}...")
-        
-        upload_qa_pairs(qa_set=qa_set, qa_pairs=chunk, prompt_column=prompt_column, completion_column=completion_column)
+def upload_qa_pairs_with_retry(qa_set, qa_pairs, retries=3, delay=5):
+    """Upload QA pairs with retry logic."""
+    for attempt in range(retries):
+        try:
+            arcee.upload_qa_pairs(qa_set=qa_set, qa_pairs=qa_pairs)
```

Comment:
Where is `arcee.upload_qa_pairs()` defined?  It looks like this function was renamed from `arcee.upload_qa_pairs()` to `upload_qa_pairs_with_retry()`, which calls a non-existent `arcee.upload_qa_pairs()`